### PR TITLE
website/docs: Update ArgoCD integration docs.

### PR DESCRIPTION
**Pull request by Gunsmithy**
## Details

Updated the ArgoCD integration documentation, primarly to add an example Terraform implementation similar to the one found in the Grafana integration docs. (Lost some time to the lack of this since I didn't know the UI defaults to having the email, openid and profile scopes when creating a new provider.)

Added an extra pointer for how to modify the ArgoCD secret when deployed with Helm, and how to add Readonly users to Argo.

Also made the opinionated change of replacing most HTTP schemes with HTTPS where they should realistically be used. :)

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [X] The documentation has been updated
-   [X] The documentation has been formatted (`make website`)

### CODE REVIEW

**Review by rissson — APPROVED** [review]
Thanks for the contribution!

File: website/integrations/services/argocd/index.md [hunk 1]
```diff
@@ -69,16 +132,25 @@ In the `argocd-secret` Secret, add the following value to the `data` field:
 dex.authentik.clientSecret: <base 64 encoded value of the Client Secret from the Provider above>
 ```
 
+If using helm, the above can be added to `configs.secret.extra` like so, securely substituting the string however you see fit:
```
Comment (tanberry) [inline website/integrations/services/argocd/index.md:135]:
```suggestion
If using Helm, the above can be added to `configs.secret.extra` as shown below, securely substituting the string however you see fit:
```

Comment (tanberry) [reply website/integrations/services/argocd/index.md:135]: I'd like to further edit this, to identify/qualify and be very specific on what "the above <?>" is... Can we say 

"If using Helm, the above string/line/definition/? can be added to a  `configs.secret.extra` section in the <__??__> file, as shown below, securely substituting the string however you see fit:"

These are nits (well not the capitalization, that's important), but for new users, it's always nice to be a bit pedantic, and say where (in what file) we are talking about.

**Review by tanberry — APPROVED** [review]
Thanks so much for this contribution @Gunsmithy ! I made a rather nit-picky request, sorry and please. :-)

File: website/integrations/services/argocd/index.md (hunk 1 above)
Comment (Gunsmithy) [reply website/integrations/services/argocd/index.md]: Ammended, see what you think!

**Review by BeryJu — APPROVED** [review]
LGTM, added one small possible improvement, but definitely not required

File: website/integrations/services/argocd/index.md [hunk 2]
```diff
@@ -46,15 +46,78 @@ Create a new _Application_ (under _Applications/Applications_) with these settin
...
+
+## Terraform provider
+
+```hcl
+data "authentik_flow" "default-provider-authorization-implicit-consent" {
+  slug = "default-provider-authorization-implicit-consent"
+}
+
+data "authentik_scope_mapping" "scope-email" {
+  name = "authentik default OAuth Mapping: OpenID 'email'"
```
[the remark this answers is not in the document]
Comment (Gunsmithy) [reply website/integrations/services/argocd/index.md:71]: Clean! I didn't realize there was a data source for these. I can make that change. I can make it on the Grafana one as well in this same PR if desired as I lifted this method from there.

[the remark this answers is not in the document]
Comment (lumhtawng2002) [reply website/integrations/services/argocd/index.md:71]: "goauthentik.io/providers/oauth2/scope-profile",

**Pull request merged**